api_compatibility_policy.md

```diff
@@ -150,10 +152,51 @@ which includes the earliest date each feature will be removed.
 
 ## Go libraries compatibility policy
 
-Tekton Pipelines may introduce breaking changes to its Go client libraries, as long as these changes
-do not impact users' yaml/json CRD definitions. For example, a change that renames a field of a CRD
-would need to follow the policy on [backwards incompatible changes](#backwards-incompatible-changes),
-but a change that renames the Go struct type for that field is allowed.
+### Covered parts of the codebase
+
+The compatibility policy for the [github.com/tektoncd/pipeline Go package](https://pkg.go.dev/github.com/tektoncd/pipeline) currently only covers
+Go structs representing CRDs in `pkg/apis`, the generated client libraries in `pkg/client`, and the [resolver interface](https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/framework/interface.go).
+Backwards compatibility for other parts of the codebase is on a best-effort basis.
+
+Pipelines contributors are working on modularizing `pkg/apis`, so that clients only need to import a minimal set of dependencies.
+This work is tracked in [issue #6483](https://github.com/tektoncd/pipeline/issues/6483).
+When it is complete, the compatibility policy will be updated to include all of `pkg/apis`.
+
+Pipelines contributors also plan to move all internal functionality in the entire [github.com/tektoncd/pipeline Go package](https://pkg.go.dev/github.com/tektoncd/pipeline)
+into internal packages. This work is tracked in [issue #5679](https://github.com/tektoncd/pipeline/issues/5679).
```

Comment:
Not related to this PR. But for the #5679,I'm worried about the benefits we could get from it outweigh the disadvantages. It's hard to find out what functions should be internal, and it also limits the growth of tekton ecosystem by disallowing the usages of current public functions

Reply:
Thanks for the feedback! I retitled the issue. I think the purpose of the issue is to move functions that we don't want to expose externally into internal packages, so that it's possible to have a compat policy for everything else. It seems like the controversial part is what should be internal and what shouldn't be, but there was general consensus on moving internal functionality into internal packages. I agree this is challenging, but it seems to me like a good long-term goal.